src/gpio.rs

```diff
@@ -123,6 +123,12 @@ pub trait TouchPin: Pin {
     fn touch_channel(&self) -> touch_pad_t;
 }
 
+/// A marker trait designating a pin which is capable of
+/// providing/receiving a rmii clk signal
+pub trait RmiiClkPin: Pin {
+    fn clock_config() -> eth_mac_clock_config_t__bindgen_ty_2;
```

Comment:
`eth_mac_clock_config_t__bindgen_ty_2` looks... yeah. Do we need *just* this part of the union, or can we switch to `eth_mac_clock_config_t`?

If we really can't how about `pub type <something more meaningful> =  eth_mac_clock_config_t__bindgen_ty_2`;

Reply:
I didn't want to copy all the contents of `eth_mac_clock_config_t` that's why I returned only the parts that change.

Reply:
`eth_mac_clock_config_t` is defined like this:
```
#[doc = " @brief Ethernet MAC Clock Configuration"]
#[doc = ""]
#[repr(C)]
#[derive(Copy, Clone)]
pub union eth_mac_clock_config_t {
    #[doc = "< EMAC MII Clock Configuration"]
    pub mii: eth_mac_clock_config_t__bindgen_ty_1,
    #[doc = "< EMAC RMII Clock Configuration"]
    pub rmii: eth_mac_clock_config_t__bindgen_ty_2,
}
#[repr(C)]
#[derive(Debug, Default, Copy, Clone)]
pub struct eth_mac_clock_config_t__bindgen_ty_1 {}
#[repr(C)]
#[derive(Debug, Copy, Clone)]
pub struct eth_mac_clock_config_t__bindgen_ty_2 {
    #[doc = "< RMII Clock Mode Configuration"]
    pub clock_mode: emac_rmii_clock_mode_t,
    #[doc = "< RMII Clock GPIO Configuration"]
    pub clock_gpio: emac_rmii_clock_gpio_t,
}
```

... so you *are* already copying "all the contents". Let's switch to `eth_mac_clock_config_t`. And let's somehow make all of this compatible with ESP-IDF 4.3. E.g. an option would be to have this trait only for ESP-IDF >= 4.4 (<=> !(ESP.IDF = 4.3)) and then in `esp-idf-svc` on V4.3 hard-code gpio0 somehow, as it used to be.

Reply:
... or if that's easier, keep the Clk trait (I like it!), but on V4.3 leave it out as a marker trait only, and then on V4.3 make gpio0 the only one implementing it.

Reply:
you are right, I was refering to `eth_mac_config_t`

Reply:
I changed it so it is only implemented for gpipo16/17 on esp idf != 4.3